pkg/cmd/cli/cmd/create/imagestream.go

```diff
@@ -0,0 +1,122 @@
+package create
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+const (
+	ImageStreamRecommendedName = "imagestream"
+
+	imageStreamLong = `
+Create a new image stream
+
+Image streams allow you to track, tag, and import images from other registries. They also define an
+access controlled destination that you can push images to.`
+
+	imageStreamExample = `  # Create a new image stream
+  %[1]s mysql`
+)
+
+type CreateImageStreamOptions struct {
+	IS     *imageapi.ImageStream
+	Client client.ImageStreamsNamespacer
+
+	Mapper       meta.RESTMapper
+	OutputFormat string
+	Out          io.Writer
+	Printer      ObjectPrinter
+}
+
+// NewCmdCreateServiceAccount is a macro command to create a new image stream
+func NewCmdCreateImageStream(name, fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	o := &CreateImageStreamOptions{Out: out}
+
+	cmd := &cobra.Command{
+		Use:     name + " NAME",
+		Short:   "Create a new empty image stream.",
+		Long:    imageStreamLong,
+		Example: fmt.Sprintf(imageStreamExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(cmd, f, args))
+			cmdutil.CheckErr(o.Validate())
+			cmdutil.CheckErr(o.Run())
+		},
+		Aliases: []string{"is"},
+	}
+
+	cmdutil.AddOutputFlagsForMutation(cmd)
+	return cmd
+}
+
+func (o *CreateImageStreamOptions) Complete(cmd *cobra.Command, f *clientcmd.Factory, args []string) error {
+	o.IS = &imageapi.ImageStream{
+		ObjectMeta: kapi.ObjectMeta{Name: args[0]},
```

Comment:
Yeah, this is a bug from the other command too.  Will fix both.

On Thu, Jun 9, 2016 at 3:43 AM, Michal Minar <notifications@github.com>
wrote:

> In pkg/cmd/cli/cmd/create/imagestream.go
> <https://github.com/openshift/origin/pull/9238#discussion_r66395598>:
>
> > +		Example: fmt.Sprintf(imageStreamExample, fullName),
> > +		Run: func(cmd *cobra.Command, args []string) {
> > +			cmdutil.CheckErr(o.Complete(cmd, f, args))
> > +			cmdutil.CheckErr(o.Validate())
> > +			cmdutil.CheckErr(o.Run())
> > +		},
> > +		Aliases: []string{"is"},
> > +	}
> > +
> > +	cmdutil.AddOutputFlagsForMutation(cmd)
> > +	return cmd
> > +}
> > +
> > +func (o *CreateImageStreamOptions) Complete(cmd *cobra.Command, f *clientcmd.Factory, args []string) error {
> > +	o.IS = &imageapi.ImageStream{
> > +		ObjectMeta: kapi.ObjectMeta{Name: args[0]},
>
> I believe the length of args needs to be checked before indexing.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9238/files/0c9ae7588c0ecba587350a2763442c64c0e4c095..00ec1e81517868c26389ccd4024d3b90ce71c62c#r66395598>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p0pmVVtjSY-wZfsqLddGYnOxvt2vks5qJ8QugaJpZM4IxjR3>
> .
>